src/core/evaluator.js

```diff
@@ -4561,7 +4561,7 @@ class EvaluatorPreprocessor {
   }
 
   static get MAX_INVALID_PATH_OPS() {
-    return shadow(this, "MAX_INVALID_PATH_OPS", 20);
+    return shadow(this, "MAX_INVALID_PATH_OPS", 10);
```

Comment:
Could you explain a bit how this relates to the lexer change below? I'm not really sure what this does.

Reply:
The first version of the patch didn't include this, but that unfortunately caused rendering regressions; please note the `bug1130815-eq-page1` regressions in https://github.com/mozilla/pdf.js/pull/14543#issuecomment-1031619775 above.
What happens is basically that the `Lexer.getNumber` changes are causing us to returning `0` in more cases than before, instead of immediately throwing, which means that rather than aborting parsing immediately we'll now continue until the following condition is satisfied: https://github.com/mozilla/pdf.js/blob/d57f3a13d7ec8d2d80cb446d55a099b06c45b047/src/core/evaluator.js#L4643-L4654

In order to work-around those problems, the only simple solution (that I found) was to reduce this constant to now accept fewer corrupt path-operators than before. Obviously all of this is *very* hand-wavy, since it's not really clear how you're supposed to (in general) deal with this type of PDF corruption; hence why I'd very much prefer not implementing https://github.com/mozilla/pdf.js/pull/14543#discussion_r800927210 as well here since that'd mean accepting "new" corrupt path-operators which could easily cause *additional* rendering regressions elsewhere.
We should probably try to *somehow* re-factor and improve all of that in the `EvaluatorPreprocessor`, but I'd prefer not doing that here.